src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -2311,6 +2311,15 @@ Node *PhiNode::Ideal(PhaseGVN *phase, bool can_reshape) {
             }
           }
         }
+
+        if (igvn) {
```

Comment:
This whole split through `mergemem` part of code is guarded by `can_reshape` check at line #2196 which is `true` only during IGVN (`PhaseIterGVN`). So you don't need to worry about executing this in parser.
But checking `igvn` value is fine here since you use it as we do at line #2278.

Reply:
Good point. I guess I could clean this up and the other `if (igvn)` checks after L2196